### PR TITLE
prevent timeout message loss when not using DTC

### PR DESCRIPTION
**Pull request by timbussmann**
## Who's affected

You might be affected if you are using one of the following transport configurations:
* RabbitMQ transport
* Azure transport 
* MSMQ or SQL transport with distributed transactions (DTC) **disabled**.

**and** you are using timeouts or deferred messages (e.g. `Bus.Defer(...)`)

## Symptoms

When experiencing connectivity issues with the transport there's a chance that a due timeout/deferred message is lost. Therefore the message will never arrive.

## What to do if you are affected

We highly recommend to update to the latest patch versions of your NServiceBus and optional persistence and transport packages. For more details see issue #2885

### CODE REVIEW

File: src/NServiceBus.AcceptanceTests/Timeouts/When_endpoint_uses_no_timeout_persistence.cs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                var settings = runDescriptor.Settings;
+
+                SetLoggingLibrary.Log4Net(null, new ContextAppender(runDescriptor.ScenarioContext, endpointConfiguration));
+
+
+                var types = GetTypesToUse(endpointConfiguration);
+
+                var config = Configure.With(types)
+                    //.DefineEndpointName(endpointConfiguration.EndpointName)
+                    .CustomConfigurationSource(configSource)
```
Comment (andreasohlund) [inline src/NServiceBus.AcceptanceTests/Timeouts/When_endpoint_uses_no_timeout_persistence.cs]: Why the commented out code?

File: src/NServiceBus.Core.Tests/Scheduler/ScheduleTests.cs [hunk 2]
```diff
@@ -36,12 +36,23 @@ public void When_scheduling_an_action_with_a_name_the_task_should_get_that_name(
         }
 
         [Test]
+        public void When_scheduling_an_anonymous_action_without_a_name_the_task_should_get_the_DeclaringType_as_name()
```
Comment (andreasohlund) [inline src/NServiceBus.Core.Tests/Scheduler/ScheduleTests.cs]: Where does this test come from? seems unrelated?

File: src/NServiceBus.Core/Persistence/Raven/TimeoutPersister/RavenTimeoutPersistence.cs [hunk 3]
```diff
@@ -171,6 +171,20 @@ public void RemoveTimeoutBy(Guid sagaId)
...
+            using (var session = OpenSession())
+            {
+                return session.Load<TimeoutData>(timeoutId);
+            }
+        }
+
+        public bool TryRemove(string timeoutId)
+        {
+            TimeoutData timeoutData;
+            return TryRemove(timeoutId, out timeoutData);
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Persistence/Raven/TimeoutPersister/RavenTimeoutPersistence.cs]: Didn't we decide to try/catch the concurrency ex and return false?

File: src/NServiceBus.Core/Timeout/Core/TimeoutPersistenceVersionCheck.cs [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        }
+
+        public static Configure SuppressOutdatedTransportWarning(this Configure configure)
+        {
+            SettingsHolder.Set(TimeoutPersistenceVersionCheck.SuppressOutdatedTransportWarning, true);
+            return configure;
+        }
+    }
+
+    internal class TimeoutPersistenceVersionCheck : IWantToRunWhenConfigurationIsComplete
```
Comment (andreasohlund) [inline src/NServiceBus.Core/Timeout/Core/TimeoutPersistenceVersionCheck.cs]: Redundant internal + move to separate file?

File: src/NServiceBus.Core/Persistence/Raven/TimeoutPersister/RavenTimeoutPersistence.cs (hunk 3 above)
Comment (timbussmann) [inline src/NServiceBus.Core/Persistence/Raven/TimeoutPersister/RavenTimeoutPersistence.cs]: good catch, that was done in the ravendb package but not here.

File: src/NServiceBus.Core.Tests/Scheduler/ScheduleTests.cs (hunk 2 above)
Comment (timbussmann) [inline src/NServiceBus.Core.Tests/Scheduler/ScheduleTests.cs]: this is due to the roslyn related changes of the declaringType. I took the fix and unit tests from v6.
